### ISSUE TITLE
Errors in parallel combinators are considered unhandled

### ISSUE THREAD

**Issue by rtimush** [zio/zio#1334]
Currently most (probably, all) parallel combinators treat errors as "unhandled", even though from the user perspective they are handled. For example, the following code prints `Left(failure)` as expected, but with the default runtime it also prints a trace for the `"failure".fail`:
```scala
"success".succeed.zipPar("failure".fail)
  .either
  .map(_.toString).flatMap(putStrLn)
```
It would be nice if ZIO understood when some other fiber would handle errors and didn't print traces in this case.

**Comment by jdegoes** [comment]
Agreed, we absolutely need this. Since most combinators just use `Fiber`, we're probably looking at a Fiber-level mechanism to basically say, "I don't care about unhandled errors in this fiber".

Note that we could already do this by adding `.ignore` to the effects before forking with them... not sure how viable that is, though. Worth a look.

**Comment by adamgfraser** [comment]
This is resolved by #1949. If we identify other combinators where spurious warnings are being reported we should open a separate issue for them and we can resolve using `forkInternal`.
